### ISSUE TITLE
Codes not compatible with IMDL-Benco

### ISSUE THREAD

**Issue by FlyTweety** [scu-zjz/SparseViT#6]
Dear authors, as you claimed that we can use IMDLBenco to this model, I have 2 questions.
1. I found the code you released is not compatible with IMDLBenco. For example, the return value of SparseViT_Mul.forward() is only a image, but in IMDLBenco, it should return a dict. Can you release the code you used in IMDLBenco ?
2. Could you also kindly provide the training script you used for training and testing with IMDLBenco so that we can accuractly and easily reproduce your result?

**Comment by DSLJDI** [comment]
Thank you for your interest! The current version we provide is solely for demonstrating the training results of SparseViT, so it only returns images. If you wish to integrate SparseViT into IMDLBenco for training, simply return the model's output (such as predicted masks, original images, losses, etc.) in a dictionary format. When we say "fully compatible," we mean that SparseViT uses the same dependency versions as IMDLBenco. You can refer to the training parameters in the paper to train SparseViT, and we will release the full training code later.

**Comment by MHpurity** [comment]
> The IMDL-BenCo itself now supports SparseViT in ! Please see https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39`model_zoo`
> 
> You can use to update the version and use to try out SparseViT training seamlessly.`benco -v``benco init model_zoo`

dear SunnyHaze:
     I can't find the evaluation file. Is this project complete?  I want to konw how I test SparseViT model and calculate AUC\F1 directly?
     Can I test SparseViT model and by cloning The IMDL-BenCo Prejrct to retrain the model again, or  should I use the pth file that I already trained from this SparseViT project and loaded into the IMDL-BenCo project for evaluation?

**Comment by SunnyHaze** [comment]
> > The IMDL-BenCo itself now supports SparseViT in ! Please see [https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39`model_zoo`](https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39%60model_zoo%60)
> > You can use to update the version and use to try out SparseViT training seamlessly.`` benco -v``benco init model_zoo ``
> 
> dear SunnyHaze: I can't find the evaluation file. Is this project complete? I want to konw how I test SparseViT model and calculate AUC\F1 directly? Can I test SparseViT model and by cloning The IMDL-BenCo Prejrct to retrain the model again, or should I use the pth file that I already trained from this SparseViT project and loaded into the IMDL-BenCo project for evaluation?

Hi MHpurity,
I think you can load the checkpoint trained from SparseViT to the IMDL-BenCo for evaluation. For cross-validation, you may test it with CASIAv1 first to check if the reported value is the same as the value in the training stage of SparseVIT project. Then you can utilize BenCo for testing on other datasets. If there are bugs or confusing outcomes, please feel free to reach out.

**Comment by MHpurity** [comment]
> > > The IMDL-BenCo itself now supports SparseViT in ! Please see [https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39`model_zoo`](https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39%60model_zoo%60)
> > > You can use to update the version and use to try out SparseViT training seamlessly.benco init model_zoo ``` benco -v`
> > 
> > 
> > dear SunnyHaze: I can't find the evaluation file. Is this project complete? I want to konw how I test SparseViT model and calculate AUC\F1 directly? Can I test SparseViT model and by cloning The IMDL-BenCo Prejrct to retrain the model again, or should I use the pth file that I already trained from this SparseViT project and loaded into the IMDL-BenCo project for evaluation?
> 
> Hi MHpurity, I think you can load the checkpoint trained from SparseViT to the IMDL-BenCo for evaluation. For cross-validation, you may test it with CASIAv1 first to check if the reported value is the same as the value in the training stage of SparseVIT project. Then you can utilize BenCo for testing on other datasets. If there are bugs or confusing outcomes, please feel free to reach out.

Thanks! I'll try

**Comment by MHpurity** [comment]
> > > The IMDL-BenCo itself now supports SparseViT in ! Please see [https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39`model_zoo`](https://github.com/scu-zjz/IMDLBenCo/releases/tag/v0.1.39%60model_zoo%60)
> > > You can use to update the version and use to try out SparseViT training seamlessly.` benco -v`benco init model_zoo ``
> > 
> > 
> > dear SunnyHaze: I can't find the evaluation file. Is this project complete? I want to konw how I test SparseViT model and calculate AUC\F1 directly? Can I test SparseViT model and by cloning The IMDL-BenCo Prejrct to retrain the model again, or should I use the pth file that I already trained from this SparseViT project and loaded into the IMDL-BenCo project for evaluation?
> 
> Hi MHpurity, I think you can load the checkpoint trained from SparseViT to the IMDL-BenCo for evaluation. For cross-validation, you may test it with CASIAv1 first to check if the reported value is the same as the value in the training stage of SparseVIT project. Then you can utilize BenCo for testing on other datasets. If there are bugs or confusing outcomes, please feel free to reach out.

Dear SunnyHaze,
I have problem on the IMDLBenCo for testing. The data  is far inferior to yours. Has my setting of the code gone wrong?

using the checkpoints.pth (https://drive.google.com/file/d/104BPPvLXkxuPu_NHaxjesdcdZ-ln92-G/view) CASIAv1 for tesing:   {"test_pixel-level F1": 0.15175569618482523, "test_pixel-level AUC": 0.5331679370092309, "epoch": 0}

My changes made on your code (IMDLBenco) :
【
def get_args_parser():
    parser = argparse.ArgumentParser('IMDLBench testing launch!', add_help=True)
    # -------------------------------
    # Model name
    parser.add_argument('--model', default='SparseViT', type=str,
                        help='The name of applied model')
    
    # 可以接受label的模型是否接受label输入，并启用相关的loss。
    parser.add_argument('--if_predict_label', action='store_true',
                        help='Does the model that can accept labels actually take label input and enable the corresponding loss function?')
    # ----Dataset parameters 数据集相关的参数----
    parser.add_argument('--image_size', default=512, type=int,
                        help='image size of the images in datasets')
    
    parser.add_argument('--if_padding', action='store_true',
                        help='padding all images to same resolution.')
    
    parser.add_argument('--if_resizing', action='store_true',
                        help='resize all images to same resolution.')
    # If edge mask activated
    parser.add_argument('--edge_mask_width', default=None, type=int,
                        help='Edge broaden size (in pixels) for edge maks generator.')
    parser.add_argument('--test_data_json', default=r'E:\PyTorch_Prj\Forgery_Detection\IMDLBenCo\test_dataset.json', type=str,
                        help='test dataset json, should be a json file contains many datasets. Details are in readme.md')
    # ------------------------------------
    # Testing 相关的参数
    parser.add_argument('--checkpoint_path', default = r'E:\PyTorch_Prj\Forgery_Detection\IMDLBenCo\IMDLBenCo\checkpoint.pth', type=str, help='path to the dir where saving checkpoints')
    parser.add_argument('--test_batch_size', default=2, type=int,
                        help="batch size for testing")
    parser.add_argument('--no_model_eval', action='store_true', 
                        help='Do not use model.eval() during testing.')

    # ----输出的日志相关的参数-----------
    parser.add_argument('--output_dir', default='./output_dir',
                        help='path where to save, empty for no saving')
    parser.add_argument('--log_dir', default='./output_dir',
                        help='path where to tensorboard log')
    # -----------------------
    
    parser.add_argument('--device', default='cuda',
                        help='device to use for training / testing')

    parser.add_argument('--num_workers', default=1, type=int)
    parser.add_argument('--pin_mem', action='store_true',
                        help='Pin CPU memory in DataLoader for more efficient (sometimes) transfer to GPU.')
    parser.add_argument('--no_pin_mem', action='store_false', dest='pin_mem')
    parser.set_defaults(pin_mem=True)

    # distributed training parameters
    parser.add_argument('--world_size', default=1, type=int,
                        help='number of distributed processes')
    parser.add_argument('--local_rank', default=-1, type=int)
    parser.add_argument('--dist_on_itp', action='store_true')
    parser.add_argument('--dist_url', default='env://',
                        help='url used to set up distributed training')
    args, remaining_args = parser.parse_known_args()
    # 获取对应的模型类
    model_class = MODELS.get(args.model)

    # 根据模型类动态创建参数解析器
    model_parser = misc.create_argparser(model_class)
    model_args = model_parser.parse_args(remaining_args)

    return args, model_args
】
Also, I use "if_resizing"
